src/profile/helpers/updateMutationVariables.ts

```diff
@@ -194,33 +199,48 @@ function formMutationArrays<T extends AddressNode | EmailNode | PhoneNode>(
   }
 }
 
-const updateMutationVariables = (
-  formValues: FormValues,
-  profile?: ProfileRoot
-): UpdateMyProfileVariables => ({
-  input: {
-    profile: {
-      firstName: formValues.firstName,
-      nickname: formValues.nickname,
-      lastName: formValues.lastName,
-      language: formValues.language,
-      ...formMutationArrays<AddressNode>(
+const updateMutationVariables: CreatePartialProfileUpdateData = (
+  formValues,
+  profile
+) => {
+  const phoneData = formValues.phones
+    ? formMutationArrays<PhoneNode>(formValues.phones, 'primaryPhone', profile)
+    : null;
+  const emailData = formValues.emails
+    ? formMutationArrays<EmailNode>(formValues.emails, 'primaryEmail', profile)
+    : null;
+  const addressData = formValues.addresses
+    ? formMutationArrays<AddressNode>(
         formValues.addresses,
         'primaryAddress',
         profile
-      ),
-      ...formMutationArrays<PhoneNode>(
-        formValues.phones,
-        'primaryPhone',
-        profile
-      ),
-      ...formMutationArrays<EmailNode>(
-        formValues.emails,
-        'primaryEmail',
-        profile
-      ),
+      )
+    : null;
+
+  const updateNameProps = _.pick(formValues, [
+    'firstName',
+    'nickname',
+    'lastName',
+  ]);
+  const updateNameCount = Object.keys(updateNameProps).length;
+
+  const userData = updateNameCount > 0 ? updateNameProps : null;
+  const additionalInformation = formValues.language
+    ? {
+        language: formValues.language,
+      }
+    : null;
+  return {
+    input: {
+      profile: {
+        ...userData,
+        ...additionalInformation,
+        ...phoneData,
+        ...emailData,
+        ...addressData,
+      },
     },
-  },
-});
+  };
+};
 
 export { formMutationArrays, updateMutationVariables };
```

Comment:
It seems in some files the exports are inlined, and in some files like this they are at the end of the file. I wonder if it would be good idea to use one style of exporting mainly? Inlined exports seems to be more common maybe?

Reply:
Also did not find usages for _formMutationArrays_. Maybe exporting it is unnecessary?

Reply:
This is an old file made by someone else. Haven't followed their preference of exporting. I think exporting where the function actually is, is much better. Won't find all occurences and change them to match that 😄